spock-core/src/main/java/org/spockframework/runtime/model/TestTag.java

```diff
@@ -0,0 +1,102 @@
+package org.spockframework.runtime.model;
+
+import org.spockframework.util.Nullable;
+
+import java.util.Objects;
+import java.util.function.IntPredicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ * A tag to for use in {@link ITestTaggable}.
+ *
+ * @since 2.2
+ * @author Leonard Brünings
+ */
+public class TestTag {
+
+  /*
+   * Should filter the same characters as JUnit Platforms test tag.
+   */
+  private static final IntPredicate INVALID_CODE_POINTS = Stream.of(
+    Character::isWhitespace,
+    Character::isISOControl,
+    isReservedChar()
+  ).reduce(IntPredicate::or).get();
+  private static final String RESERVED_CHARACTERS = "!&(),|";
+  private final String value;
+
+  private TestTag(String value) {
+    this.value = value;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  /**
+   * Creates a new {@link TestTag} with the given value.
+   *
+   * Must not contain any of the following characters:
+   * <ul>
+   *   <li>White Space</li>
+   *   <li>ISO Control Characters</li>
+   *   <li>{@code !}</li>
+   *   <li>{@code &}</li>
+   *   <li>{@code (}</li>
+   *   <li>{@code )}</li>
+   *   <li>{@code ,}</li>
+   *   <li>{@code |}</li>
+   * </ul>
+   *
+   * @param tag the value of the tag
+   * @return a new {@link TestTag}
+   */
+  public static TestTag create(String tag) {
+    if(!isValid(tag)){
+      throw new IllegalArgumentException("Invalid tag: [" + tag + "] contains invalid character(s): " + getInvalidCharacters(tag));
+    }
+    return new TestTag(tag);
+  }
+
+
+  public static boolean isValid(@Nullable String value) {
+    return value != null && value.length() > 0 && value.codePoints().noneMatch(INVALID_CODE_POINTS);
+  }
+
+  private static String getInvalidCharacters(@Nullable String value) {
```

Comment:
I think we should move this to the JUnit Platform so this `TestTag` can be a simple wrapper around JUnit Platform's `TestTag`.

Reply:
Should I open a PR?